lib/studies/shrine/app.dart

```diff
@@ -112,32 +112,36 @@ class _ShrineAppState extends State<ShrineApp> with TickerProviderStateMixin {
       ),
     );
 
-    return ScopedModel<AppStateModel>(
-      model: _model,
-      child: WillPopScope(
-        onWillPop: _onWillPop,
-        child: MaterialApp(
-          title: 'Shrine',
-          debugShowCheckedModeBanner: false,
-          initialRoute: ShrineApp.loginRoute,
-          onGenerateInitialRoutes: (_) {
-            return [
-              MaterialPageRoute<void>(
-                builder: (context) => const LoginPage(),
-              ),
-            ];
-          },
-          routes: {
-            ShrineApp.loginRoute: (context) => const LoginPage(),
-            ShrineApp.homeRoute: (context) => home,
-          },
-          theme: shrineTheme.copyWith(
-            platform: GalleryOptions.of(context).platform,
+    return RestorationScope(
+      restorationId: 'shrineAppState',
+      child: ScopedModel<AppStateModel>(
+        model: _model,
+        child: WillPopScope(
+          onWillPop: _onWillPop,
+          child: MaterialApp(
+            restorationScopeId: 'shrineApp',
+            title: 'Shrine',
+            debugShowCheckedModeBanner: false,
+            initialRoute: ShrineApp.loginRoute,
+            onGenerateInitialRoutes: (_) {
+              return [
+                MaterialPageRoute<void>(
+                  builder: (context) => const LoginPage(),
```

Comment:
The problem here is that the route(s) returned by `onGenerateInitialRoutes` do not have `Route.settings.name` set. This means, they cannot be restored and you end up with an empty set of routes to restore. We should provide a better error here. I'll file an issue for it.

In this case, the fix is really simple: Just remove the `onGenerateInitialRoutes` argument altogether. The app is already specifying an `initialRoute: ShrineApp.loginRoute`, which when looked up in the `routes` table below produces a route with a LoginPage - that's exactly the same thing that onGenerateInitialRoutes is doing. So instead of specifying it, we can just rely on the defaults.

Reply:
Filed https://github.com/flutter/flutter/issues/72453.